### DISCUSSION TITLE
Starlette

### DISCUSSION THREAD

**Discussion by dineshbvadhia** [NRWLDev/fastapi-problem#22] [Q&A] [answered]
Could fastapi-problem be made to work with a Starlette only application?

**Comment by EdgyEdgemond** [comment]
You would need to create a different library, but essentially removing the fastapi.RequestValidationError handler from this code base would work with starlette.

The old version of the library used to have support for starlette, but was removed as it made no sense to use this with starlette and install fastapi unnecessarily.

As well as other fastapi imports.

**Reply by EdgyEdgemond** [reply]
https://github.com/NRWLDev/fastapi-problem/pull/23
This should be the code required to have a working starlette version of this library.

**Reply by dineshbvadhia** [reply]
Oh thank-you for showing how to do a starlette version. The reason I asked is because though a fastapi-based application-server has been built, I'm coming to think fastapi is overkill. Again, thanks. Will let you know if I go down the starlette route.

**Reply by EdgyEdgemond** [reply]
let me know if you think a starlette version would be useful, can create that and expand on it for fastapi without too much effort without duplicating much code

**Reply by dineshbvadhia** [reply]
A three-part answer:

1. I started using a FastApi RFC7807 package a couple of years ago but the software wasn't documented well nor maintained. So, I didn't use it optimally unlike fastapi-problem for which I have received terrific help.

2. Standard and consistent http error handling and formatting with RFC9457 should be a must for all http-based systems. Individually and collectively the benefits would be enormous. It would also make interoperability easier. 

3. FastApi (built on Starlette and Pydantic) and Starlette are independent http-based packages. So, my vote would be to support both.

**Reply by EdgyEdgemond** [reply, marked as answer]
https://pypi.org/project/starlette-problem/

Try this one out :)
